### PR TITLE
Fixes for Python 3.6

### PR DESCRIPTION
**Pull request by besser82**
This fixes the testsuite on Python 3.6.

### CODE REVIEW

**Comment by besser82** [comment]
Should be ready for merging, now.  Could you tag a new release, please?

File: testfixtures/tests/test_should_raise.py [hunk 1]
```diff
@@ -191,8 +192,12 @@ def test_import_errors_1(self):
             message = "No module named 'textfixtures'"
         else:
             message = 'No module named textfixtures.foo.bar'
-        with ShouldRaise(ImportError(message)):
-            import textfixtures.foo.bar
+        if sys.version_info[:2] >= (3,6):
```
Comment (cjw296) [inline testfixtures/tests/test_should_raise.py]: Please introduce a new `py_36_plus constant` and then simplify such that:

```
exception = ModuleNotFoundError if py_36_plus else ImportError
with ShouldRaise(exception(message)):
```

File: testfixtures/tests/test_shouldwarn.py [hunk 2]
```diff
@@ -106,14 +106,24 @@ def test_maximal_explore(self):
...
-                  category=DeprecationWarning,
-                  file=None,
-                  filename='bar.py',
-                  line=None,
-                  lineno=42,
-                  message=C(DeprecationWarning('foo'))
-                  ), recorded[0])
-
-
+        if sys.version_info[:2] >= (3,6):
```
Comment (cjw296) [inline testfixtures/tests/test_shouldwarn.py]: use the constant again here, and then I'd suggest:
```
expected_attrs = dict(
    _category_name='DeprecationWarning',
    category=DeprecationWarning,
    file=None,
    filename='bar.py',
    line=None,
    lineno=42,
    message=C(DeprecationWarning('foo')),
)
if py_36_plus:
    expected_attrs['source'] = None
compare(expected=C(warnings.WarningMessage, **expected_attrs),
        actual=recorded[0])
```

**Comment by besser82** [comment]
Your requested changes are in the rebased commits.  Well, I don't _need_ a new version, but it would be nice to indicate on PYPI that this is compatible with Python 3.6, too…

**Comment by cjw296** [comment]
Fantastic, thanks! The reason I ask is that I have a big branch I need to land:

https://github.com/cjw296/testfixtures/tree/pytest

...moving to pytest for the test runner and from Manuel to Sybil for testing docs. I still need to write the docs for Sybil and then release before testfixtures can have a new release depending on that.

So, there will be a release, but it's going to be a bigger one in maybe a week or two's time.
Thanks again for your work!

**Comment by besser82** [comment]
Thank you for merging!  =)

**Pull request merged**